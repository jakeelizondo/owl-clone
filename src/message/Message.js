import React from 'react';

function Message(props) {
  if (props.type === 'message') {
    return (
      <div className="chat-message">
        <img className="chat-avatar" src={props.avatar} />
        <p>{props.name}</p>
        <p>{props.time}</p>
        <p>{props.message}</p>
      </div>
    );
  } else if (props.type === 'thumbs-up' || props.type === 'thumbs-down') {
    return (
      <div>
        <p>
          {props.name} gave a {props.type}
        </p>
      </div>
    );
  } else if (props.type === 'raise-hand') {
    return (
      <div>
        <p>{props.name} raised their hand</p>
      </div>
    );
  } else if (props.type === 'clap') {
    return (
      <div>
        <p>{props.name} clapped</p>
      </div>
    );
  } else if (props.type.includes('join')) {
    return (
      <div>
        <p>{props.name} joined the stage</p>
      </div>
    );
  } else if (props.type.includes('leave')) {
    return (
      <div>
        <p>{props.name} left the stage</p>
      </div>
    );
  }
}

export default Message;
