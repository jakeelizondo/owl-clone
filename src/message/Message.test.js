import React from 'react';
import ReactDOM from 'react-dom';
import Message from './Message';
import renderer from 'react-test-renderer';

describe('Message Component', () => {
  // smoke test for app component

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Message
        avatar="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
        message="hello"
        name="my name"
        time="10:47"
        type="message"
      />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  // snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <Message
          avatar="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
          message="hello"
          name="my name"
          time="10:47"
          type="message"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
