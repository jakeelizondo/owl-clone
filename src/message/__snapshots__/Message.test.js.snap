// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Component renders the UI as expected 1`] = `
<div
  className="chat-message"
>
  <img
    className="chat-avatar"
    src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
  />
  <p>
    my name
  </p>
  <p>
    10:47
  </p>
  <p>
    hello
  </p>
</div>
`;
